frontend/react/style-guide.md

```diff
@@ -37,23 +37,37 @@ src
 └───app
 │   │
 │   └───components
-│   │ └───baseComponents
-│   │     └───Input
-│   │     └───Text
-│   │     └───Button
+│   │ └───atoms # Can't be broken down into smaller things
+│   │ |   └───Input
+│   │ |   └───Text
+│   │ |   └───Button
+│   │ |   └───etc
+│   │ └───molecules # Two or more atoms
+│   │ |   └───FormInput
+│   │ |   └───SearchBar
+│   │ |   └───Table
+│   │ |   └───etc
+│   │ └───organisms # Two or more molecules
+│   │     └───LoginForm
+│   │     └───UserTable
 │   │     └───etc
 │   └───screens
 │       └───MyScreenComponent
-│           └───assets // Screen specific app assets
-│           | components
+│           └───assets # Screen specific app assets
+│           └───reducer
+│             | actions.js
+│             | reducer.js
+│             | selectors.js
+│             | effects.js
```

Comment:
Do you think it is a good practice to have the reducer and actions in each component? Maybe i'm not seeing the advantages, but i think that having the actions and reducer here will grow very quickly.
When having 2 or more screens that need the same action in which action do we add it? Also I think this makes it much more difficult to find specific actions and/or services.
Another thing i don't think is ver nice is that in the store we will have to import screens to generate the store.
I feel that having this practice has more disadvantages than advantages.

Reply:
I agree with Lucas, i think that at some point we will have reference problems if we move screens or components to another place

Reply:
Yeah ok, this is one thing that I've been thinking of for a while and will be addressed in the Advanced Redux talk we'll have in some weeks. But the idea is to mainly shift the criteria for creating reducers from "models" (Product, Pizza, Car, etc) to screens (ProductList, Menu, CarDetail, etc) and only leave "model" reducers for stuff that will really be global (like User or some global configuration we fetch from API).
Having only "model" reducers may have the following issues:
1) One screen changing the store may, by mistake, influence another by changing a global state (for example, if ProductList changes `products` in Product reducer and there's another screen with a different product list that uses the same field in the store) If you don't clean the store correctly you'll have issues.
2) "model" reducers tend to be large and difficult to refactor, especially if its a core model of the app and it's used in a lot of screens in different ways.
3) Having "screen" reducers makes it easier to refactor to/from Context+useReducer which in some cases it's a better solution than redux
4) There's a lot of data in model reducers that will be changed only by one screen, which makes it hard to read when debugging one particular thing. Also, knowing which screen changed some shared field like `products` may be problematic in some cases

Having said all this, I propose we start using both (model and screen reducers), which may need a better analysis of which data will be global and which won't. 
Answering your question, if two screens need the same action is because they're modifying a shared state. This may mean one of the following:
1) That state should be global, not in a specific screen
2) The reducer logic may be extracted as an effect and reused in each screen's reducer, we're duplicating the action but we're reusing the effect and gaining readability I think.

Reply:
Okey, i think it is a good idea. What i would recommend is to place all the reducers in the redux folder. And to keep them with the same name as the screen. I don't think importing the reducers from the screens to create the store is a good practice. However all of this can change if we start using context, in that case i completely agree of having the reducers in the screen folder. Because you will use it only there.

Reply:
wouldn't it be the same? The only difference is that you import in store.js from the screens, but aside from that unavoidable difference, shouldn't we handle the "state management" folders the same way either if it's redux or context+reducer what we're using?

Reply:
I think the main difference is that with a store you can access all the reducers, you can be in one screen and access the state of another-one. I know we are trying to avoid that, but you can still do it. Meanwhile with context you cannot access a context of another screen because the context is the father of the screen index. I think that is the main reason i think we should treat this two state management separately. Also... i don't like importing screens for their reducers to create the store.